### PR TITLE
feat: add isMocked method to check if the member of the object is mocked

### PR DESCRIPTION
**Pull request by popomore**

### CODE REVIEW

**Comment by popomore** [comment]
@fent

**Comment by fent** [comment]
Hi. This looks really good for a PR, with tests even. 😮 

I only have a few minor issues with this

* I'd use `assert.ok()` instead of `assert.equal()` when expecting a boolean.
* Not a fan of assigning another property to the mocked object. I know that there's a very low chance that the user might actually be using that key themselves, specially if it was changed from "muk_" to something less likely. But I'm not sure. Searching the list of `mocks` for the object and key is another way to do this, but it could be non-performant if the user uses `muk.isMocked()` a lot and has mocked several things. What do you think?
* Needs to be added to readme. Understandable that it's not, since I've only placed examples there and not clear list of the API. I could see it going under the `muk.restore()` example.

**Comment by popomore** [comment]
I try 3 ways to check if it is mocked, the main thing is where is the flag.

1. The way in this PR, I think it's a easiest way.

1. The way that cache the object and key in the hash table, Map is a good choice that can set the key as object but Object can't. But Map is supported after Node@0.12.

1. The way that loop all mocks to match object and key every calling isMocked, I concern about the performance.

So I choose the first way, if we can make a break change that drop before Node@0.12, I can try the second way.

**Comment by alsotang** [comment]
@fent the usage of `assert` is little question...both are ok..

and I think add a no-enumerate property to a object is acceptable. perhaps we can use `__muk_` prefix instead of `muk_`.

@popomore Map is a good choice right. But as a inner implement, can we also use Array and indexOf to do this purpose?

**Comment by alsotang** [comment]
By the way, I +1 for this pr.

**Comment by fent** [comment]
I wasn't aware Map existed. Node v0.10 is not a hard requirement, it would be ok to remove that from `.travis.yml`.

**Comment by popomore** [comment]
I have updated this PR using Map and Set. Delete the 0.10 from travis.

File: lib/method.js [hunk 1]
```diff
@@ -69,3 +77,9 @@ method.restore = function restoreMocks() {
   requireMocks = [];
   */
 };
+
+method.isMocked = function isMocked(obj, key) {
+  var flag = cache.get(obj);
+  if (!flag) return false;
```
Comment (fent) [inline lib/method.js]: This project has curly brackets around if statements, even one liners. Or you could also do `return flag && flag.has(key);`

File: lib/method.js [hunk 2]
```diff
@@ -39,8 +40,15 @@ var method = module.exports = function mockMethod(obj, key, method) {
     enumerable: true,
     value: method
   });
-};
 
+  // set a flag that check if it is mocked
```
Comment (fent) [inline lib/method.js]: checks*

**Comment by popomore** [comment]
Updated.

**Comment by fent** [comment]
Thanks for the quality PR. Published v0.5.0

**Pull request merged**